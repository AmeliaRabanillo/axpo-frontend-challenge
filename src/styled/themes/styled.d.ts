import 'styled-components';

declare module 'styled-components' {
    export interface DefaultTheme {
        fontFamily: string;
        fontSizes: {
            small: number;
            medium: number;
            large: number;
        };
        fontMeasureUnit: string;
        colors: {
            primary: string;
            black: string;
            white: string;
        };
    }
}