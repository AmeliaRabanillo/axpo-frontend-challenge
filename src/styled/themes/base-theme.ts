// @ts-ignore
import {DefaultTheme} from 'styled-components';

export const baseTheme: DefaultTheme = {
    fontFamily: '"Helvetica", sans-serif',
    fontSizes: {
        small: 14,
        medium: 16,
        large: 24,
    },
    fontMeasureUnit: 'px',

    colors: {
        primary: '#ff5d64',
        black: '#3d3d3d',
        white: '#f5e9d2',
    }
};
