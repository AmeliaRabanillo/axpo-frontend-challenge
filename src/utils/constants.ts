enum SearchTypeEnum {
    POP_DENSITY = 'population_density',
    DRONE_RES = 'drone_restriction',
}

const SEARCH_TYPE_DISPLAY_MAP = {
    [SearchTypeEnum.POP_DENSITY]: 'Population Density',
    [SearchTypeEnum.DRONE_RES]: 'Drone Restrictions'
};


enum RoutesEnum {
    START_PAGE = '/',
    POPULATION_DENSITY_TABLE_PAGE = '/population_density',
    DRONE_RESTRICTION_TABLE_PAGE = '/drone_restriction',
}

const MAP_SERVER_URL = 'https://api3.geo.admin.ch/rest/services/api/MapServer/identify';
const DRONE_RESTRICTIONS_LAYER = 'all:ch.bazl.einschraenkungen-drohnen';
const POPULATION_DENSITY_LAYER = 'all:ch.bfs.volkszaehlung-bevoelkerungsstatistik_einwohner';
const MAP_SERVER_GEOMETRY_TYPE = 'esriGeometryPoint';
const MAP_SERVER_SR = '4326';

export {
    SearchTypeEnum,
    RoutesEnum,
    SEARCH_TYPE_DISPLAY_MAP,
    MAP_SERVER_URL,
    DRONE_RESTRICTIONS_LAYER,
    POPULATION_DENSITY_LAYER,
    MAP_SERVER_GEOMETRY_TYPE,
    MAP_SERVER_SR,
}