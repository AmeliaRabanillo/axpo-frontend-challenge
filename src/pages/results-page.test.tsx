import React from 'react'
import 'jest-styled-components';
import {ThemeProvider} from "styled-components";
import {baseTheme} from "../styled/themes/base-theme";
import {DRONE_RESTRICTIONS_LAYER, SearchTypeEnum} from "../utils/constants";
import {act} from "@testing-library/react";
import ResultsPage from "./results-page";
import mockAxios from "jest-mock-axios";
import ReactDOM from 'react-dom/client';

const mockedUseNavigate = jest.fn();
const mockedUseLocation = jest.fn();
var mockUseLocationResult = {
    state: {
        id: '1',
        name: 'fake-name',
        latitude: '2',
        longitude: '3',
        type: 'fake-type'
    }
}
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom') as any,
    useNavigate: () => mockedUseNavigate,
    useLocation: () => mockUseLocationResult,
}));

let windowSpy = jest.spyOn(window, 'alert').mockImplementation(() => {
});

describe("Result page tests", () => {

    // @ts-ignore
    let container;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
    });

    afterEach(() => {
        mockedUseNavigate.mockClear();
        mockedUseLocation.mockClear();
        // @ts-ignore
        document.body.removeChild(container);
        container = null;
    })

    test("Result Page for drone restrictions matches snapshot", async () => {
        const fakeResponse = {
            results:
                [{
                    "layerBodId": DRONE_RESTRICTIONS_LAYER,
                    "layerName": "Population (residents)",
                    "featureId": 4625514,
                    "id": 4625514,
                    "attributes": {
                        "number": 7,
                        "i_year": 2010,
                        "reli": 75602210,
                        "label": 75602210
                    }
                },
                ]
        }
        // @ts-ignore
        mockAxios.get.mockImplementationOnce(() => new Promise((res, rej) => res({data: fakeResponse})));

        await act(() => {
            // @ts-ignore
            ReactDOM.createRoot(container).render(<ThemeProvider theme={baseTheme}>
                    <ResultsPage type={SearchTypeEnum.DRONE_RES}/>
                </ThemeProvider>
            );
        });
        // @ts-ignore
        expect(container).toMatchSnapshot();
        expect(windowSpy).not.toHaveBeenCalled();
    });

    test("Result Page for drone restrictions service errors matches snapshot", async () => {
        // @ts-ignore
        mockAxios.get.mockImplementationOnce(() => new Promise((res, rej) => rej(new TypeError('This is a random error'))));
        jest.spyOn(window, 'alert').mockImplementation(() => {
        });

        await act(() => {
            // @ts-ignore
            ReactDOM.createRoot(container).render(<ThemeProvider theme={baseTheme}>
                    <ResultsPage type={SearchTypeEnum.DRONE_RES}/>
                </ThemeProvider>
            );
        });
        // @ts-ignore
        expect(container).toMatchSnapshot();
        expect(windowSpy).toHaveBeenCalledTimes(1);
    });

    test("Result Page for population density matches snapshot", async () => {
        const fakeResponse = {
            results:
                [{
                    "layerBodId": DRONE_RESTRICTIONS_LAYER,
                    "layerName": "Population (residents)",
                    "featureId": 4625514,
                    "id": 4625514,
                    "attributes": {
                        "number": 7,
                        "i_year": 2010,
                        "reli": 75602210,
                        "label": 75602210
                    }
                },
                ]
        }
        // @ts-ignore
        mockAxios.get.mockImplementationOnce(() => new Promise((res, rej) => res({data: fakeResponse})));

        await act(() => {
            // @ts-ignore
            ReactDOM.createRoot(container).render(<ThemeProvider theme={baseTheme}>
                <ResultsPage type={SearchTypeEnum.POP_DENSITY}/>
            </ThemeProvider>);
        });

        // @ts-ignore
        expect(container).toMatchSnapshot();
        expect(windowSpy).not.toHaveBeenCalled();
    });

    test("Result Page for population density service errors matches snapshot", async () => {
        // @ts-ignore
        mockAxios.get.mockImplementationOnce(() => new Promise((res, rej) => rej(new TypeError('This is a random error'))));
        jest.spyOn(window, 'alert').mockImplementation(() => {
        });

        await act(() => {
            // @ts-ignore
            ReactDOM.createRoot(container).render(<ThemeProvider theme={baseTheme}>
                <ResultsPage type={SearchTypeEnum.POP_DENSITY}/>
            </ThemeProvider>);
        });
        // @ts-ignore
        expect(container).toMatchSnapshot();
        expect(windowSpy).toHaveBeenCalledTimes(1);
    });

    test("Result Page with unknown search type matches snapshot", async () => {

        await act(() => {
            // @ts-ignore

            ReactDOM.createRoot(container).render(
                <ThemeProvider theme={baseTheme}>
                    {/*@ts-ignore*/}
                    <ResultsPage type='fake type'/>
                </ThemeProvider>);
        });
        // @ts-ignore
        expect(container).toMatchSnapshot();
        expect(windowSpy).toHaveBeenCalledTimes(1);
    });

});