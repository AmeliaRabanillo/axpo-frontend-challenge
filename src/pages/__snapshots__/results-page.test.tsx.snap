// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result page tests Result Page for drone restrictions matches snapshot 1`] = `
.c1 {
  color: #f5e9d2;
  background-color: #ff5d64;
  font-size: 16px;
  margin: 20px;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 25px;
}

.c1 span {
  margin-right: 12px;
}

.c1:hover {
  cursor: pointer;
}

.c2 {
  border-radius: 50%;
  padding: 8px;
  margin-left: 0;
}

.c3 {
  color: #3d3d3d;
  font-size: 32px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  color: #3d3d3d;
  margin: 8px 0;
}

<div>
  <div>
    <div
      class="c0"
    >
      <button
        class="c1 c2"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="KeyboardBackspaceIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
          />
        </svg>
      </button>
      <h2
        class="c3"
      >
        Here are your results for 
        Drone Restrictions
      </h2>
    </div>
    <p
      class="c4"
    >
      <b>
        Id:
      </b>
       
      1
    </p>
    <p
      class="c4"
    >
      <b>
        Name:
      </b>
       
      fake-name
    </p>
    <p
      class="c4"
    >
      <b>
        Type:
      </b>
       
      fake-type
    </p>
    <p
      class="c4"
    >
      <b>
        Latitude:
      </b>
       
      2
    </p>
    <p
      class="c4"
    >
      <b>
        Longitude:
      </b>
       
      3
    </p>
    <div
      class="_2IvMF _GzYRV"
    >
      <div
        class="_2bkNM"
        role="list"
      >
        <span
          class="_16FAf _f10Tu _1MFti"
          role="button"
          tabindex="0"
        >
          ▾
        </span>
        <span
          class="_3uHL6 _3eOF8"
        >
          {
        </span>
        <div>
          <div
            class="_2bkNM"
            role="list"
          >
            <span
              class="_16FAf _f10Tu _1MFti"
              role="button"
              tabindex="0"
            >
              ▾
            </span>
            <span
              class="_1MGIk"
            >
              results
              :
            </span>
            <span
              class="_3uHL6 _3eOF8"
            >
              [
            </span>
            <div>
              <div
                class="_2bkNM"
                role="list"
              >
                <span
                  class="_16FAf _f10Tu _1MFti"
                  role="button"
                  tabindex="0"
                >
                  ▾
                </span>
                <span
                  class="_3uHL6 _3eOF8"
                >
                  {
                </span>
                <div>
                  <div
                    class="_2bkNM"
                    role="listitem"
                  >
                    <span
                      class="_1MGIk"
                    >
                      layerBodId
                      :
                    </span>
                    <span
                      class="_vGjyY"
                    >
                      "all:ch.bazl.einschraenkungen-drohnen"
                    </span>
                    <span
                      class="_3uHL6 _3eOF8"
                    >
                      ,
                    </span>
                  </div>
                  <div
                    class="_2bkNM"
                    role="listitem"
                  >
                    <span
                      class="_1MGIk"
                    >
                      layerName
                      :
                    </span>
                    <span
                      class="_vGjyY"
                    >
                      "Population (residents)"
                    </span>
                    <span
                      class="_3uHL6 _3eOF8"
                    >
                      ,
                    </span>
                  </div>
                  <div
                    class="_2bkNM"
                    role="listitem"
                  >
                    <span
                      class="_1MGIk"
                    >
                      featureId
                      :
                    </span>
                    <span
                      class="_1bQdo"
                    >
                      4625514
                    </span>
                    <span
                      class="_3uHL6 _3eOF8"
                    >
                      ,
                    </span>
                  </div>
                  <div
                    class="_2bkNM"
                    role="listitem"
                  >
                    <span
                      class="_1MGIk"
                    >
                      id
                      :
                    </span>
                    <span
                      class="_1bQdo"
                    >
                      4625514
                    </span>
                    <span
                      class="_3uHL6 _3eOF8"
                    >
                      ,
                    </span>
                  </div>
                  <div
                    class="_2bkNM"
                    role="list"
                  >
                    <span
                      class="_16FAf _f10Tu _1MFti"
                      role="button"
                      tabindex="0"
                    >
                      ▾
                    </span>
                    <span
                      class="_1MGIk"
                    >
                      attributes
                      :
                    </span>
                    <span
                      class="_3uHL6 _3eOF8"
                    >
                      {
                    </span>
                    <div>
                      <div
                        class="_2bkNM"
                        role="listitem"
                      >
                        <span
                          class="_1MGIk"
                        >
                          number
                          :
                        </span>
                        <span
                          class="_1bQdo"
                        >
                          7
                        </span>
                        <span
                          class="_3uHL6 _3eOF8"
                        >
                          ,
                        </span>
                      </div>
                      <div
                        class="_2bkNM"
                        role="listitem"
                      >
                        <span
                          class="_1MGIk"
                        >
                          i_year
                          :
                        </span>
                        <span
                          class="_1bQdo"
                        >
                          2010
                        </span>
                        <span
                          class="_3uHL6 _3eOF8"
                        >
                          ,
                        </span>
                      </div>
                      <div
                        class="_2bkNM"
                        role="listitem"
                      >
                        <span
                          class="_1MGIk"
                        >
                          reli
                          :
                        </span>
                        <span
                          class="_1bQdo"
                        >
                          75602210
                        </span>
                        <span
                          class="_3uHL6 _3eOF8"
                        >
                          ,
                        </span>
                      </div>
                      <div
                        class="_2bkNM"
                        role="listitem"
                      >
                        <span
                          class="_1MGIk"
                        >
                          label
                          :
                        </span>
                        <span
                          class="_1bQdo"
                        >
                          75602210
                        </span>
                      </div>
                    </div>
                    <span
                      class="_3uHL6 _3eOF8"
                    >
                      }
                    </span>
                  </div>
                </div>
                <span
                  class="_3uHL6 _3eOF8"
                >
                  }
                </span>
              </div>
            </div>
            <span
              class="_3uHL6 _3eOF8"
            >
              ]
            </span>
          </div>
        </div>
        <span
          class="_3uHL6 _3eOF8"
        >
          }
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Result page tests Result Page for drone restrictions service errors matches snapshot 1`] = `
.c1 {
  color: #f5e9d2;
  background-color: #ff5d64;
  font-size: 16px;
  margin: 20px;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 25px;
}

.c1 span {
  margin-right: 12px;
}

.c1:hover {
  cursor: pointer;
}

.c2 {
  border-radius: 50%;
  padding: 8px;
  margin-left: 0;
}

.c3 {
  color: #3d3d3d;
  font-size: 32px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  color: #3d3d3d;
  margin: 8px 0;
}

<div>
  <div>
    <div
      class="c0"
    >
      <button
        class="c1 c2"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="KeyboardBackspaceIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
          />
        </svg>
      </button>
      <h2
        class="c3"
      >
        Here are your results for 
        Drone Restrictions
      </h2>
    </div>
    <p
      class="c4"
    >
      <b>
        Id:
      </b>
       
      1
    </p>
    <p
      class="c4"
    >
      <b>
        Name:
      </b>
       
      fake-name
    </p>
    <p
      class="c4"
    >
      <b>
        Type:
      </b>
       
      fake-type
    </p>
    <p
      class="c4"
    >
      <b>
        Latitude:
      </b>
       
      2
    </p>
    <p
      class="c4"
    >
      <b>
        Longitude:
      </b>
       
      3
    </p>
    <p>
      This is a random error
    </p>
  </div>
</div>
`;

exports[`Result page tests Result Page for population density matches snapshot 1`] = `
.c1 {
  color: #f5e9d2;
  background-color: #ff5d64;
  font-size: 16px;
  margin: 20px;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 25px;
}

.c1 span {
  margin-right: 12px;
}

.c1:hover {
  cursor: pointer;
}

.c2 {
  border-radius: 50%;
  padding: 8px;
  margin-left: 0;
}

.c3 {
  color: #3d3d3d;
  font-size: 32px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  color: #3d3d3d;
  margin: 8px 0;
}

<div>
  <div>
    <div
      class="c0"
    >
      <button
        class="c1 c2"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="KeyboardBackspaceIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
          />
        </svg>
      </button>
      <h2
        class="c3"
      >
        Here are your results for 
        Population Density
      </h2>
    </div>
    <p
      class="c4"
    >
      <b>
        Id:
      </b>
       
      1
    </p>
    <p
      class="c4"
    >
      <b>
        Name:
      </b>
       
      fake-name
    </p>
    <p
      class="c4"
    >
      <b>
        Type:
      </b>
       
      fake-type
    </p>
    <p
      class="c4"
    >
      <b>
        Latitude:
      </b>
       
      2
    </p>
    <p
      class="c4"
    >
      <b>
        Longitude:
      </b>
       
      3
    </p>
    <div
      class="_2IvMF _GzYRV"
    >
      <div
        class="_2bkNM"
        role="list"
      >
        <span
          class="_16FAf _f10Tu _1MFti"
          role="button"
          tabindex="0"
        >
          ▾
        </span>
        <span
          class="_3uHL6 _3eOF8"
        >
          {
        </span>
        <div>
          <div
            class="_2bkNM"
            role="list"
          >
            <span
              class="_16FAf _f10Tu _1MFti"
              role="button"
              tabindex="0"
            >
              ▾
            </span>
            <span
              class="_1MGIk"
            >
              results
              :
            </span>
            <span
              class="_3uHL6 _3eOF8"
            >
              [
            </span>
            <div>
              <div
                class="_2bkNM"
                role="list"
              >
                <span
                  class="_16FAf _f10Tu _1MFti"
                  role="button"
                  tabindex="0"
                >
                  ▾
                </span>
                <span
                  class="_3uHL6 _3eOF8"
                >
                  {
                </span>
                <div>
                  <div
                    class="_2bkNM"
                    role="listitem"
                  >
                    <span
                      class="_1MGIk"
                    >
                      layerBodId
                      :
                    </span>
                    <span
                      class="_vGjyY"
                    >
                      "all:ch.bazl.einschraenkungen-drohnen"
                    </span>
                    <span
                      class="_3uHL6 _3eOF8"
                    >
                      ,
                    </span>
                  </div>
                  <div
                    class="_2bkNM"
                    role="listitem"
                  >
                    <span
                      class="_1MGIk"
                    >
                      layerName
                      :
                    </span>
                    <span
                      class="_vGjyY"
                    >
                      "Population (residents)"
                    </span>
                    <span
                      class="_3uHL6 _3eOF8"
                    >
                      ,
                    </span>
                  </div>
                  <div
                    class="_2bkNM"
                    role="listitem"
                  >
                    <span
                      class="_1MGIk"
                    >
                      featureId
                      :
                    </span>
                    <span
                      class="_1bQdo"
                    >
                      4625514
                    </span>
                    <span
                      class="_3uHL6 _3eOF8"
                    >
                      ,
                    </span>
                  </div>
                  <div
                    class="_2bkNM"
                    role="listitem"
                  >
                    <span
                      class="_1MGIk"
                    >
                      id
                      :
                    </span>
                    <span
                      class="_1bQdo"
                    >
                      4625514
                    </span>
                    <span
                      class="_3uHL6 _3eOF8"
                    >
                      ,
                    </span>
                  </div>
                  <div
                    class="_2bkNM"
                    role="list"
                  >
                    <span
                      class="_16FAf _f10Tu _1MFti"
                      role="button"
                      tabindex="0"
                    >
                      ▾
                    </span>
                    <span
                      class="_1MGIk"
                    >
                      attributes
                      :
                    </span>
                    <span
                      class="_3uHL6 _3eOF8"
                    >
                      {
                    </span>
                    <div>
                      <div
                        class="_2bkNM"
                        role="listitem"
                      >
                        <span
                          class="_1MGIk"
                        >
                          number
                          :
                        </span>
                        <span
                          class="_1bQdo"
                        >
                          7
                        </span>
                        <span
                          class="_3uHL6 _3eOF8"
                        >
                          ,
                        </span>
                      </div>
                      <div
                        class="_2bkNM"
                        role="listitem"
                      >
                        <span
                          class="_1MGIk"
                        >
                          i_year
                          :
                        </span>
                        <span
                          class="_1bQdo"
                        >
                          2010
                        </span>
                        <span
                          class="_3uHL6 _3eOF8"
                        >
                          ,
                        </span>
                      </div>
                      <div
                        class="_2bkNM"
                        role="listitem"
                      >
                        <span
                          class="_1MGIk"
                        >
                          reli
                          :
                        </span>
                        <span
                          class="_1bQdo"
                        >
                          75602210
                        </span>
                        <span
                          class="_3uHL6 _3eOF8"
                        >
                          ,
                        </span>
                      </div>
                      <div
                        class="_2bkNM"
                        role="listitem"
                      >
                        <span
                          class="_1MGIk"
                        >
                          label
                          :
                        </span>
                        <span
                          class="_1bQdo"
                        >
                          75602210
                        </span>
                      </div>
                    </div>
                    <span
                      class="_3uHL6 _3eOF8"
                    >
                      }
                    </span>
                  </div>
                </div>
                <span
                  class="_3uHL6 _3eOF8"
                >
                  }
                </span>
              </div>
            </div>
            <span
              class="_3uHL6 _3eOF8"
            >
              ]
            </span>
          </div>
        </div>
        <span
          class="_3uHL6 _3eOF8"
        >
          }
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Result page tests Result Page for population density service errors matches snapshot 1`] = `
.c1 {
  color: #f5e9d2;
  background-color: #ff5d64;
  font-size: 16px;
  margin: 20px;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 25px;
}

.c1 span {
  margin-right: 12px;
}

.c1:hover {
  cursor: pointer;
}

.c2 {
  border-radius: 50%;
  padding: 8px;
  margin-left: 0;
}

.c3 {
  color: #3d3d3d;
  font-size: 32px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  color: #3d3d3d;
  margin: 8px 0;
}

<div>
  <div>
    <div
      class="c0"
    >
      <button
        class="c1 c2"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="KeyboardBackspaceIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
          />
        </svg>
      </button>
      <h2
        class="c3"
      >
        Here are your results for 
        Population Density
      </h2>
    </div>
    <p
      class="c4"
    >
      <b>
        Id:
      </b>
       
      1
    </p>
    <p
      class="c4"
    >
      <b>
        Name:
      </b>
       
      fake-name
    </p>
    <p
      class="c4"
    >
      <b>
        Type:
      </b>
       
      fake-type
    </p>
    <p
      class="c4"
    >
      <b>
        Latitude:
      </b>
       
      2
    </p>
    <p
      class="c4"
    >
      <b>
        Longitude:
      </b>
       
      3
    </p>
    <p>
      This is a random error
    </p>
  </div>
</div>
`;

exports[`Result page tests Result Page with unknown search type matches snapshot 1`] = `
.c1 {
  color: #f5e9d2;
  background-color: #ff5d64;
  font-size: 16px;
  margin: 20px;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 25px;
}

.c1 span {
  margin-right: 12px;
}

.c1:hover {
  cursor: pointer;
}

.c2 {
  border-radius: 50%;
  padding: 8px;
  margin-left: 0;
}

.c3 {
  color: #3d3d3d;
  font-size: 32px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  color: #3d3d3d;
  margin: 8px 0;
}

<div>
  <div>
    <div
      class="c0"
    >
      <button
        class="c1 c2"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="KeyboardBackspaceIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 11H6.83l3.58-3.59L9 6l-6 6 6 6 1.41-1.41L6.83 13H21z"
          />
        </svg>
      </button>
      <h2
        class="c3"
      >
        Here are your results for 
      </h2>
    </div>
    <p
      class="c4"
    >
      <b>
        Id:
      </b>
       
      1
    </p>
    <p
      class="c4"
    >
      <b>
        Name:
      </b>
       
      fake-name
    </p>
    <p
      class="c4"
    >
      <b>
        Type:
      </b>
       
      fake-type
    </p>
    <p
      class="c4"
    >
      <b>
        Latitude:
      </b>
       
      2
    </p>
    <p
      class="c4"
    >
      <b>
        Longitude:
      </b>
       
      3
    </p>
    <p>
      Unknown type to search
    </p>
  </div>
</div>
`;
